haskell/defs.bzl

```diff
@@ -294,6 +290,26 @@ def haskell_binary(
         **kwargs
     )
 
+    repl_kwargs = {
+        attr: kwargs[attr]
+        for attr in ["testonly", "tags"]
+        if attr in kwargs
+    }
+    native.alias(
```

Comment:
Do we really care about backward compatibility here?

It won't break dramatically and if the release note contains the right lines, the migration is trivial.

Reply:
We've been getting issues referring to the `@repl` targets, and they are documented in a couple places. So, removing them would likely affect a good number of users. The goal of this PR is only to change implementation, not interface.

Reply:
Thank you.